### ISSUE TITLE
can FFX learn the relationship within time series? 

### ISSUE THREAD

**Issue by isaac-you** [natekupp/ffx#34]
Thank you for your great work which offers new approach for symbolic regression. By your example in the slice, FFX can output the function of  linear or nonlinear expression. But when I study the stock time series, I usually need auto-relation function expression. For example: **ts_min(x, d) = time-series min over the past d days** .  **ts_argmin(x, d) = which day ts_min(x, d) occurred on** . **ts_rank(x, d) = time-series rank in the past d days** . **stddev(x, d) = moving time-series standard deviation over the past d days**. 

So  I have problems to output the function express above , can you give me some advice , thank you for your help. @jmmcd

**Comment by jmmcd** [comment]
I think it's natural to make a new matrix with columns x, ts_min(x, d), ts_argmin(x, d), ts_rank(x, d), etc. I would do this as pre-processing before using FFX.

**Comment by isaac-you** [comment]
I don't think this can work, because those function cannot be determined at first, we hope that the FFX can output the function expression

**Comment by jmmcd** [comment]
The problem isn't clear to me. 

> those function cannot be determined at first

I think I can just use `np.min()`. What am I missing?
